### PR TITLE
EWPP-3194: Various UX improvements.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: modules/oe_translation_epoetry/src/Plugin/views/field/TargetLanguagesWithTooltip.php [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      $count += count($grouped[TranslationRequestEpoetryInterface::STATUS_LANGUAGE_REVIEW]);
+    }
+
+    // Include the synced ones.
+    if (isset($grouped[TranslationRequestEpoetryInterface::STATUS_LANGUAGE_SYNCHRONISED])) {
+      $count += count($grouped[TranslationRequestEpoetryInterface::STATUS_LANGUAGE_SYNCHRONISED]);
+    }
+
+    // Include also the languages which have been accepted locally, but only
+    // if the there is translation data.
```
Comment (22Alexandra) [inline modules/oe_translation_epoetry/src/Plugin/views/field/TargetLanguagesWithTooltip.php]:
```suggestion
    // if there is translation data.
```

File: modules/oe_translation_epoetry/src/Plugin/views/field/TargetLanguagesWithTooltip.php [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+   *
+   * @param \Drupal\oe_translation_epoetry\TranslationRequestEpoetryInterface $request
+   *   The request.
+   * @param string $language
+   *   The language.
+   *
+   * @return bool
+   *   Whether a given translation has arrived from ePoetry.
+   */
+  protected function isLanguageArrived(TranslationRequestEpoetryInterface $request, string $language): bool {
```
Comment (22Alexandra) [inline modules/oe_translation_epoetry/src/Plugin/views/field/TargetLanguagesWithTooltip.php]:
```suggestion
  protected function hasLanguageArrived(TranslationRequestEpoetryInterface $request, string $language): bool {
```

File: modules/oe_translation_epoetry/src/Plugin/views/filter/TargetLanguagesFilter.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, Connection $connection, LanguageManagerInterface $languageManager) {
```
Comment (22Alexandra) [inline modules/oe_translation_epoetry/src/Plugin/views/filter/TargetLanguagesFilter.php]: What do we need the database for? Doesn't seem to be used.

**Review by 22Alexandra — COMMENTED** [review]
The view link of the ongoing remote translations leads to the request's canonical page and we cannot go back to the dashboard from there. I think we should use the `entity.$entity_type_id.remote_translation` route.
view link

**Comment by upchuk** [comment]
@22Alexandra 

> The view link of the ongoing remote translations leads to the request's canonical page and we cannot go back to the dashboard from there. I think we should use the entity.$entity_type_id.remote_translation route.

There can be multiple requests on the remote_translation route so we cannot link there always. However, if you enable breakdcrumbs in your theme, you can use those to navigate (check in ewcms)

**Review by hernani — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**